### PR TITLE
An/12 testing array schema serialization

### LINKED ISSUE #12: Add unit tests for array schema serialization
Add unit tests for array schema serialization. Currently there is an [unit-capi-array_schema.cc](https://github.com/Shelnutt2/TileDB/blob/dev/test/src/unit-capi-array_schema.cc) which has various tests for array schemas. We need to modify this test to add support for serialization. You should be able to create shims for the various array schema operations like @tdenniston did for dense.

Here is the PR where @tdenniston add shims for serialization so we can use a single dense unit test query file but test both serialize and none serialize version: https://github.com/Shelnutt2/TileDB/pull/13/files

First I'd look at `unit-capi-array_schema.cc` and create a list of the functions you think we should add shims to to test the serialization. For querying it was just `submit` and `finalize`, for array schema it will probably be `load`, `create`, etc.

After `unit-capi-array_schema.cc` is updated we can safely remove the `unit-capi-capnp.cc` file as it will be superseeded.

### PR DESCRIPTION
**Pull request by antalakas**
Adds unit tests for array schema serialization.
Closes #12

### CODE REVIEW

File: test/src/unit-capi-array_schema.cc [hunk 1]
```diff
@@ -94,6 +95,12 @@ struct ArraySchemaFx {
   const char* DIM2_TILE_EXTENT_STR = "5";
   const uint64_t TILE_EXTENT_SIZE = sizeof(TILE_EXTENTS) / DIM_NUM;
 
+  /**
+   * If true, queries are serialized before submission, to test the
+   * serialization paths.
+   */
+  bool serialize_query = false;
```
Comment (Shelnutt2) [inline test/src/unit-capi-array_schema.cc]: Can you change this to serialize_array_schema? Also update the comment

**Review by Shelnutt2 — APPROVED** [review]
Looks good, one minor comment. Also can you squash the commits into 1 with a clean commit message?
